src/tree/utils/populateTree.ts

```diff
@@ -245,7 +246,22 @@ function prepareSchema(
   try {
     return resolveSchema(schema, path, options);
   } catch (ex) {
-    (node as TreeListParentNode).children = [];
-    return void (node as TreeListParentNode).children.push(createErrorTreeNode(node as TreeListParentNode, ex.message));
+    const treeNode: SchemaTreeListNode = {
+      id: generateId(),
+      name: '',
+      parent: node as TreeListParentNode,
+      children: [],
+    };
+
+    (node as TreeListParentNode).children.push(treeNode);
+    metadataStore.set(treeNode, {
+      schemaNode: processNode(schema || {}),
+      schema: schema || {},
+      path,
+    });
+
+    treeNode.children.push(createErrorTreeNode(treeNode, ex.message));
```

Comment:
Can this go right into the `treeNode` declaration? (as in `children: [createEr....]`)

Or does the `metadataStore.set` call mutate `treeNode`?

Reply:
That might not work as `createErrorTreeNode` extracts metadata from parent which is `treeNode`. So it needs to have that metadata set first.

Reply:
Oh I see. Not obvious but makes sense. 👍